powerhome/playbook
# Better error messages, and more test coverage

#### Screens

See added tests and updated error messages. Specifically the additions to `props_spec.rb` and `props/base_spec.rb`. The other changes in prop type tests are simply to keep them working.

#### Breaking Changes

No breaking changes.

#### Runway Ticket URL

[CMPN-143](https://nitro.powerhrg.com/runway/backlog_items/CMPN-143)

#### Other

Consider merging with the next major release if you like. Should be a nice developer experience enhancement.

#### How to test this

1. Make sure milano envs loads normally. (It does)
2. After it's merged, see the error become more informative locally and caught in sentry. [Here is an example of what the error message looked like before this change](https://sentry.io/organizations/power-home-remodeling-group/issues/1918768597/?query=is%3Aunresolved+%22required+prop+and+needs+to+be+provided+a+value%22&statsPeriod=14d) .  

```
#<Playbook::Props::String:0x000000001129fa48 @default=nil, @required=true> is a required prop and needs to be provided a value
```

And after:

```
PbExampleKit prop 'necessary_prop' of type String is required and needs a value
```
and
```
PbExampleKit has invalid value of '"wrong_value"' for prop 'test_prop' of type Boolean
```

#### Checklist:

- [x] **LABELS** Add a label: `enhancement`, `bug`, `improvement`, `new kit`, `depreciation`, or `breaking`. See [Changelog & Labels](https://github.com/powerhome/playbook/wiki/Changelog-&-Labels) for details.
- [x] **SCREENSHOT** Please add a screen shot or two. (See the tests that expose the error message.)

*The normal release cut off deadline is 3p EDT each week. Please reach out to the release team if you have an urgent request or need a release off cycle.*
